Sources/Sentry/SentryFileManager.m

```diff
@@ -174,11 +181,41 @@ - (NSString *)storeEnvelope:(SentryEnvelope *)envelope
     @synchronized(self) {
         NSString *result = [self storeData:[SentrySerialization dataWithEnvelope:envelope error:nil]
                                     toPath:self.envelopesPath];
-        [self handleFileManagerLimit:self.envelopesPath maxCount:self.maxEnvelopes];
+        [self handleEnvelopesLimit];
```

Comment:
I believe we should run this before storing the file otherwise we'll first 1 file too many and delete one as a result.

Reply:
That is a good point, that I didn't consider. The downside with running it before would be that we could lose this envelope while handling the limit when we have many envelopes and the app crashes in the meantime. Furthermore, the logic handling the limit gets more complex, because we also have to consider the envelope in memory in `SentryMigrateSessionInit` and store it to disk if it contains the session to migrate the init flag. 
To keep it simple we could just keep the number of envelopes at 99 in `handleEnvelopesLimit` so we make sure we don't have more than 100 envelopes or we just keep it as it is, because the number is internal and the main purpose of it is that our SDK doesn't use up too much disk space and it can be that 50 envelopes use up more space than 100, but we would need to add this info as a comment. What do you think?

Reply:
sounds good to me